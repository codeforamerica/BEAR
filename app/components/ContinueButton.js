import React, { Component } from 'react';

type Props = {
  onContinue: () => {},
  disabled: boolean
};

export default class ContinueButton extends Component<Props> {
  classes = () => {
    const buttonClasses = 'button button--primary';

    const { disabled } = this.props;
    if (disabled) {
      return buttonClasses.concat(' button--disabled');
    }
    return buttonClasses;
  };

  render() {
    const { onContinue, disabled } = this.props;
    return (
      <div>
        <button
          className={this.classes()}
          onClick={onContinue}
          type="button"
          id="continue"
          disabled={disabled}
        >
          Continue <i className="icon-arrow_forward" />
        </button>
      </div>
    );
  }
}
