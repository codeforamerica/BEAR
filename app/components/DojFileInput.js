// @flow
import React, { Component } from 'react';
import styles from './DOJFileInput.css';

type Props = {
  onFileSelect: string => void
};

export default class DojFileInput extends Component<Props> {
  handleFileSelection = (event: SyntheticEvent<HTMLInputElement>) => {
    const { onFileSelect } = this.props;
    const filePath = event.currentTarget.files[0].path;
    onFileSelect(filePath);
  };

  render() {
    return (
      <div className="file-upload">
        <label className="button file-upload__label" htmlFor="doj-file-input">
          <input
            onChange={this.handleFileSelection}
            type="file"
            accept=".dat, .csv"
            id="doj-file-input"
            name="doj-file-input"
            className="file-upload__input"
          />
          Select file
        </label>
        <p className={styles.emptyMessage}>No file selected</p>
      </div>
    );
  }
}
