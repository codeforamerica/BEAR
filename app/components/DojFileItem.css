.fileName {
  display: flex;
  align-items: center;
}

.fileRemove {
  padding-left: 5px;
  cursor: pointer;
}
