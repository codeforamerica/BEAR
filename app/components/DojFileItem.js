// @flow
import React, { Component } from 'react';
import path from 'path';
import styles from './DojFileItem.css';

type Props = {
  filePath: string,
  onFileRemove: string => void
};

export default class DojFileItem extends Component<Props> {
  formatFileName = () => {
    const { filePath } = this.props;
    return filePath.split(path.sep).pop();
  };

  fileRemove = () => {
    const { filePath, onFileRemove } = this.props;
    onFileRemove(filePath);
  };

  render() {
    return (
      <div className="doj-file">
        <p className={styles.fileName}>
          File imported: {this.formatFileName()}
          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/interactive-supports-focus */}
          <i
            className={`${styles.fileRemove} icon-close`}
            onClick={this.fileRemove}
            role="button"
          />
        </p>
      </div>
    );
  }
}
