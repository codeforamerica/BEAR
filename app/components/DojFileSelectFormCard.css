.semibold {
  font-weight: 500;
}

.outlineBox {
  padding: 20px;
  border: 2px solid #f7f5f4;
}
