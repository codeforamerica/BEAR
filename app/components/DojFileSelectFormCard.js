// @flow
import React, { Component } from 'react';
import DojFileInput from './DojFileInput';
import ContinueButton from './ContinueButton';
import DojFileItem from './DojFileItem';

import FormCard, {
  FormCardContent,
  FormCardFooter,
  FormCardHeader
} from './FormCard';

type Props = {
  currentScreen: number,
  dojFilePath: string,
  updateFilePath: string => void
};

const screenNumber = 2;

export default class DojFileSelectFormCard extends Component<Props> {
  renderContinueButton = () => {
    const { dojFilePath } = this.props;
    if (dojFilePath !== '') {
      return <ContinueButton onContinue={() => {}} />;
    }
  };

  renderCardContent = () => {
    const { dojFilePath, updateFilePath } = this.props;
    if (dojFilePath !== '') {
      return (
        <DojFileItem filePath={dojFilePath} onFileRemove={updateFilePath} />
      );
    }

    return <DojFileInput onFileSelect={updateFilePath} />;
  };

  render() {
    const { currentScreen } = this.props;
    return (
      <FormCard currentScreen={currentScreen} screenNumber={screenNumber}>
        <FormCardHeader>Upload .dat file</FormCardHeader>
        <FormCardContent>{this.renderCardContent()}</FormCardContent>
        <FormCardFooter>{this.renderContinueButton()}</FormCardFooter>
      </FormCard>
    );
  }
}
