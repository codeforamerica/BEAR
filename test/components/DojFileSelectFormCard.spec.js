import sinon from 'sinon';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import DojFileSelectFormCard from '../../app/components/DojFileSelectFormCard';
import ContinueButton from '../../app/components/ContinueButton';
import DojFileItem from '../../app/components/DojFileItem';
import DojFileInput from '../../app/components/DojFileInput';

Enzyme.configure({ adapter: new Adapter() });
const sandbox = sinon.sandbox.create();

function setup(dojFilePath) {
  const fakeUpdateFilePath = sandbox.spy();
  const component = mount(
    <DojFileSelectFormCard
      currentScreen={2}
      updateFilePath={fakeUpdateFilePath}
      dojFilePath={dojFilePath}
    />
  );
  return {
    component,
    fakeUpdateFilePath
  };
}

afterEach(() => {
  sandbox.restore();
});

describe('DojFileSelectFormCard component', () => {
  describe('the Upload button', () => {
    it('should appear if the file path is empty', () => {
      const { component } = setup('');
      expect(component.containsAnyMatchingElements([<DojFileInput />])).toEqual(
        true
      );
    });

    it('should NOT appear if the file path is not empty', () => {
      const { component } = setup('path/to/file');
      expect(component.containsAnyMatchingElements([<DojFileInput />])).toEqual(
        false
      );
    });
  });

  describe('the Continue button', () => {
    it('should not appear if the file path is empty', () => {
      const { component } = setup('');
      expect(
        component.containsAnyMatchingElements([<ContinueButton />])
      ).toEqual(false);
    });

    it('should appear if the file path is not empty', () => {
      const { component } = setup('path/to/file');
      expect(
        component.containsAnyMatchingElements([<ContinueButton />])
      ).toEqual(true);
    });
  });

  describe('the file name', () => {
    it('should not appear if the file path is empty', () => {
      const { component } = setup('');
      expect(component.containsAnyMatchingElements([<DojFileItem />])).toEqual(
        false
      );
    });

    it('should appear if the file path is not empty', () => {
      const { component } = setup('path/to/file');
      expect(component.containsAnyMatchingElements([<DojFileItem />])).toEqual(
        true
      );
    });
  });

  it('should match exact snapshot when file has not been selected', () => {
    const component = (
      <div>
        <DojFileSelectFormCard currentScreen={2} dojFilePath="" />
      </div>
    );

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should match exact snapshot when file has been selected', () => {
    const component = (
      <div>
        <DojFileSelectFormCard currentScreen={2} dojFilePath="path/to/file" />
      </div>
    );

    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
