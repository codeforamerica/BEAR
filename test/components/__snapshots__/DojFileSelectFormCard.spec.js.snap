// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DojFileSelectFormCard component should match exact snapshot when file has been selected 1`] = `
<div>
  <div
    className="form-card"
  >
    <header
      className="form-card__header"
    >
      <h1
        className="form-card__title"
      >
        Upload .dat file
      </h1>
    </header>
    <div
      className="form-card__content"
    >
      <div
        className="doj-file"
      >
        <p
          className="fileName"
        >
          file
          <i
            className="fileRemove icon-close"
            onClick={[Function]}
            role="button"
          />
        </p>
      </div>
    </div>
    <div
      className="form-card__footer"
    >
      <div>
        <button
          className="button button--primary"
          onClick={[Function]}
          type="button"
        >
          Continue 
          <i
            className="icon-arrow_forward"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`DojFileSelectFormCard component should match exact snapshot when file has not been selected 1`] = `
<div>
  <div
    className="form-card"
  >
    <header
      className="form-card__header"
    >
      <h1
        className="form-card__title"
      >
        Upload .dat file
      </h1>
    </header>
    <div
      className="form-card__content"
    >
      <div
        className="file-upload"
      >
        <label
          className="button file-upload__label"
          htmlFor="doj-file-input"
        >
          <input
            className="file-upload__input"
            id="doj-file-input"
            name="doj-file-input"
            onChange={[Function]}
            type="file"
          />
          Upload File
        </label>
      </div>
    </div>
    <div
      className="form-card__footer"
    />
  </div>
</div>
`;
